modules/dop/providers/devworkflow/devworkflow.service.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/erda-project/erda/modules/dop/providers/devworkflow/db"
+	"github.com/erda-project/erda/modules/dop/services/apierrors"
+	"github.com/erda-project/erda/pkg/common/apis"
+)
+
+const resource = "devWorkflow"
+
+type Service interface {
+	CreateDevWorkflow(context.Context, *pb.CreateDevWorkflowRequest) (*pb.CreateDevWorkflowResponse, error)
+	DeleteDevWorkflow(context.Context, *pb.DeleteDevWorkflowRequest) (*pb.DeleteDevWorkflowResponse, error)
+	UpdateDevWorkflow(context.Context, *pb.UpdateDevWorkflowRequest) (*pb.UpdateDevWorkflowResponse, error)
+	GetDevWorkflowsByProjectID(context.Context, *pb.GetDevWorkflowRequest) (*pb.GetDevWorkflowResponse, error)
+}
+
+type ServiceImplement struct {
+	db  *db.Client
+	bdl *bundle.Bundle
+}
+
+func (s *ServiceImplement) CreateDevWorkflow(ctx context.Context, request *pb.CreateDevWorkflowRequest) (*pb.CreateDevWorkflowResponse, error) {
+	project, err := s.bdl.GetProject(request.ProjectID)
+	if err != nil {
+		return nil, apierrors.ErrCreateDevWorkflow.InternalError(err)
+	}
+	org, err := s.bdl.GetOrg(project.OrgID)
+	if err != nil {
+		return nil, apierrors.ErrCreateDevWorkflow.InternalError(err)
+	}
+
+	flows := s.InitWorkFlows()
+	b, err := json.Marshal(&flows)
+	if err != nil {
+		return nil, apierrors.ErrCreateDevWorkflow.InternalError(err)
+	}
+	wf := db.DevWorkflow{
+		Scope: db.Scope{
+			OrgID:       org.ID,
+			OrgName:     org.Name,
+			ProjectID:   project.ID,
+			ProjectName: project.Name,
+		},
+		Operator: db.Operator{
+			Creator: request.UserID,
+			Updater: request.UserID,
+		},
+		WorkFlows: b,
+	}
+	err = s.db.CreateWf(&wf)
+	if err != nil {
+		return nil, apierrors.ErrCreateDevWorkflow.InternalError(err)
+	}
+	return &pb.CreateDevWorkflowResponse{Data: wf.Convert()}, nil
+}
+
+func (s *ServiceImplement) InitWorkFlows() db.WorkFlows {
+	return db.WorkFlows{
+		{
+			Name:             "DEV",
+			FlowType:         "two_branch",
+			TargetBranch:     "develop",
+			ChangeFromBranch: "",
+			ChangeBranch:     "feature/*,bugfix/*",
+			EnableAutoMerge:  true,
+			AutoMergeBranch:  "dev/temp/*",
+			Artifact:         "alpha",
+			Environment:      "DEV",
+			StartWorkflowHints: []db.StartWorkflowHint{
+				{
+					Place:            "TASK",
+					ChangeBranchRule: "feature/*",
+				},
+				{
+					Place:            "BUG",
+					ChangeBranchRule: "bugfix/*",
+				},
+			},
+		},
+		{
+			Name:               "TEST",
+			FlowType:           "single_branch",
+			TargetBranch:       "develop",
+			ChangeFromBranch:   "",
+			ChangeBranch:       "",
+			EnableAutoMerge:    false,
+			AutoMergeBranch:    "",
+			Artifact:           "beta",
+			Environment:        "TEST",
+			StartWorkflowHints: nil,
+		},
+		{
+			Name:               "STAGING",
+			FlowType:           "three_branch",
+			TargetBranch:       "master",
+			ChangeFromBranch:   "develop",
+			ChangeBranch:       "release/*",
+			EnableAutoMerge:    true,
+			AutoMergeBranch:    "staging/temp/*",
```

Comment:
`STAGING` flow should `EnableAutoMerge: false` and have no AutoMergeBranch

Reply:
done